variables.tf

```diff
@@ -1,36 +1,109 @@
+  default     = []
+  description = "List of event streams target to be created"
+  sensitive   = true
+
   validation {
-    condition     = contains(["standard", "cos-one-rate-plan"], var.plan)
-    error_message = "The specified pricing plan is not available. The following plans are supported: 'standard', 'cos-one-rate-plan'"
+    condition     = alltrue([for es_target in var.eventstreams_targets : (es_target.service_to_service_enabled == true && es_target.api_key == null) || (es_target.service_to_service_enabled == false && es_target.api_key != null)])
+    error_message = "If 'service_to_service_enabled' is true, 'api_key' value should not be passed. If you wish to use 'api_key', set 'service_to_service_enabled' to false."
   }
 }
 
-variable "resource_group_id" {
-  type        = string
-  description = "The ID of the resource group where you want to create the service."
+# Cloud Logs Targets
+variable "cloud_logs_targets" {
+  type = list(object({
+    instance_id                              = string
+    target_region                            = optional(string)
+    target_name                              = string
+    skip_atracker_cloud_logs_iam_auth_policy = optional(bool, false)
+  }))
+  default     = []
+  description = "List of Cloud Logs targets to be created"
 }
 
-variable "resource_tags" {
-  type        = list(string)
-  description = "List of resource tag to associate with the instance."
+# Routes
+variable "activity_tracker_routes" {
+  type = list(object({
+    locations  = list(string)
+    target_ids = list(string)
+    route_name = string
+  }))
+  description = "List of routes to be created, maximum four routes are allowed"
   default     = []
+
+  validation {
+    condition     = length(var.activity_tracker_routes) <= 4
+    error_message = "Number of routes should be less than or equal to 4"
+  }
+
+  validation {
+    condition     = alltrue([for activity_tracker_route in var.activity_tracker_routes : length(activity_tracker_route.locations) > 0])
+    error_message = "Length of locations can not be zero"
+  }
+
+  validation {
+    condition     = alltrue([for activity_tracker_route in var.activity_tracker_routes : length(activity_tracker_route.target_ids) > 0])
+    error_message = "Length of target_ids can not be zero"
+  }
+}
+# Event Routing Setting
+variable "global_event_routing_settings" {
+  type = object({
+    default_targets           = optional(list(string), [])
+    metadata_region_primary   = string
+    metadata_region_backup    = optional(string)
+    permitted_target_regions  = list(string)
+    private_api_endpoint_only = optional(bool, false)
+  })
+  description = "Global settings for event routing"
+  default     = null
+
+  # https://cloud.ibm.com/docs/atracker?topic=atracker-regions#regions-atracker
+  validation {
+    error_message = "Valid regions for permitted_target_regions are: us-south, eu-de, us-east, eu-es, eu-gb, au-syd, br-sao, ca-tor, eu-es, jp-tok, jp-osa, in-che, eu-fr2"
```

Comment:
make it more clear.
`Valid regions for permitted_target_regions` -> `Valid regions for permitted_target_regions of .... are`